grakn-dist/src/grakn

```diff
@@ -0,0 +1,662 @@
+    status=0 # storage running
+  else
+    status=1 # storage not running
+  fi
+  return $status
+}
+
+storage_wait_until_ready() {
+  local now_s=`date '+%s'`
+  local stop_s=$(( $now_s + $STORAGE_STARTUP_TIMEOUT_S ))
+  local status_thrift=
+
+  while [ $now_s -le $stop_s ]; do
+      echo -n "."
+      # The \r\n deletion bit is necessary for Cygwin compatibility
+      status_thrift=`"${GRAKN_HOME}"/services/cassandra/nodetool statusthrift 2>/dev/null | tr -d '\n\r'`
+      if [ $? -eq 0 -a 'running' = "$status_thrift" ]; then
+          return 0
+      fi
+      sleep $WAIT_INTERVAL_S
+      now_s=`date '+%s'`
+  done
+
+  return 1
+}
+
+storage_start_and_wait_until_ready() {
+  storage_is_storage_running
+  is_storage_running=$?
+
+  if [ $is_storage_running -eq 0 ]; then
+    echo "Storage is already running"
+    return 0 # storage is already running. no need to do anything
+  else
+    # otherwise attempt to start
+    echo -n "Starting Storage..."
+    storage_start_storage
+    local exec_storage_bin=$?
+
+    # check storage status and wait until ready
+    local status=
+    if [ $exec_storage_bin -eq 0 ] ; then
+      # exec storage bin successful. now wait until ready
+      storage_wait_until_ready
+      status=$?
+      echo "SUCCESS"
+    else
+      # exec storage bin failed
+      status=1
+      echo "FAILED!"
+    fi
+
+    return $status
+  fi
+}
+
+grakn_server_stop_storage() {
+  echo -n "Stopping Storage..."
+  storage_is_storage_running
+  local is_storage_running=$?
+  if [[ $is_storage_running -eq 0 ]]; then
+    if [[ -e $STORAGE_PID ]]; then
+      pid=`cat $STORAGE_PID`
+      kill "$pid"
+      while kill -0 "$pid" 2>/dev/null; do
+          sleep 0.5
+      done
+      echo "SUCCESS"
+      return 0
+    else
+      echo "FAILED!" # unable to find PID
+      return 1
+    fi
+  else
+    echo "NOT RUNNING"
+  fi
+}
+
+storage_clean_storage() {
+  rm -rf db/cassandra
+  mkdir -p db/cassandra/data db/cassandra/commitlog db/cassandra/saved_caches
+}
+
+# ================================================
+# queue helper functions
+# ================================================
+queue_start_queue() {
+  local queue_bin=""
+  if [ "$(uname)" == "Darwin" ]; then
+      queue_bin="redis-server-osx"
+  elif [ "$(expr substr $(uname -s) 1 5)" == "Linux" ]; then
+      queue_bin="redis-server-linux"
+  fi
+
+  # run queue
+  # queue needs to be ran with $GRAKN_HOME as the working directory
+  # otherwise it won't be able to find its data directory located at $GRAKN_HOME/db/redis
+  pushd "$GRAKN_HOME" > /dev/null
+  "${GRAKN_HOME}"/services/redis/$queue_bin "${GRAKN_HOME}"/services/redis/redis.conf
+  local status=$?
+  popd > /dev/null
+
+  return $status
+}
+
+queue_is_queue_running() {
+  local status=
+  local count_running_queue_process=`ps -ef | grep 'redis-server' | grep -v grep | awk '{ print $2}' | wc -l`
+  if [ $count_running_queue_process -gt 0 ] ; then
+    status=0 # queue running
```

Comment:
instead of 0 and 1, can we use running and stopped for readability further down?